iModelCore/ECDb/ECDb/ECSql/ECSqlParser.cpp

```diff
@@ -4070,6 +4170,91 @@ BentleyStatus ECSqlParser::ParseWindowFrameExclusion(WindowFrameClauseExp::Windo
+        return ERROR;
+        }
+
+    switch (parseNode->getChild(0)->getTokenID())
+        {
+        case SQL_TOKEN_NAVIGATION_VALUE:
+            {
+            std::unique_ptr<NavValueCreationFuncExp> navValueCreationFuncExp = nullptr;
+            if (SUCCESS != ParseNavValueCreationFuncExp(navValueCreationFuncExp, parseNode))
+                return ERROR;
+            valueExp = std::move(navValueCreationFuncExp);
+            return SUCCESS;
+            }
+        default:
+            {
+            BeAssert(false && "Unhandled tokenID in ValueCreationFuncExp");
+            return ERROR;
+            }
+        }
+    }
+
+//-----------------------------------------------------------------------------------------
+// @bsimethod
+//+---------------+---------------+---------------+---------------+---------------+--------
+BentleyStatus ECSqlParser::ParseNavValueCreationFuncExp(std::unique_ptr<NavValueCreationFuncExp>& valueCreationFuncExp, OSQLParseNode const* parseNode) const
+    {
+    std::unique_ptr<DerivedPropertyExp> derivedPropertyExp = nullptr;
+    if (SUCCESS != ParseDerivedColumn(derivedPropertyExp, parseNode->getChild(2)))
+        return ERROR;
+
+    if (derivedPropertyExp->GetExpression()->GetType() != Exp::Type::PropertyName) // NAVIGATION_VALUE function should accept only PropertyName
+        return ERROR;
+
+    if (derivedPropertyExp->GetExpression()->GetAs<PropertyNameExp>().GetPropertyPath().Size() != 3)
+        return ERROR;
+
+    PropertyPath propPath = derivedPropertyExp->GetExpression()->GetAs<PropertyNameExp>().GetPropertyPath();
+
+    ClassMap const* classMap = m_context->GetECDb().Schemas().GetDispatcher().GetClassMap(propPath[0].GetName(), propPath[1].GetName(), SchemaLookupMode::AutoDetect, nullptr);
+
+    if (classMap == nullptr)
+        return ERROR;
+
+    propPath.SetClassMap(*classMap);
+    std::shared_ptr<ClassNameExp::Info> classNameExpInfo = nullptr;
+    if (SUCCESS != m_context->TryResolveClass(classNameExpInfo, nullptr, propPath[0].GetName(), propPath[1].GetName(), ECSqlType::Select, false, *parseNode))
+        return ERROR;
+
+    std::unique_ptr<ClassNameExp> classNameExp =  std::make_unique<ClassNameExp>(propPath[1].GetName().c_str(), propPath[0].GetName().c_str(), nullptr, classNameExpInfo);
+
+    if (classMap->GetPropertyMaps().Find(propPath[2].GetName().c_str()) == nullptr)
+        return ERROR;
+
+    std::unique_ptr<PropertyNameExp> propNameExp = std::make_unique<PropertyNameExp>(
+        *m_context,
+        propPath[2].GetName(),
+        *classNameExp.get(),
+        *classMap
+    );
+
+    propNameExp->SetTypeInfo(ECSqlTypeInfo(ECSqlTypeInfo::Kind::Navigation));
+    Utf8CP alias = parseNode->getParent()->getChild(1)->getTokenValue().empty() ? nullptr : parseNode->getParent()->getChild(1)->getTokenValue().c_str();
+    derivedPropertyExp = std::move(std::make_unique<DerivedPropertyExp>(std::move(propNameExp), alias));

```

Comment:
Compiler error from LLVM (Linux): mmoving a temporary object prevents copy elision

Reply:
> std::move(std::make_unique<
That is wrong. Get rid of std::move()

Reply:
done